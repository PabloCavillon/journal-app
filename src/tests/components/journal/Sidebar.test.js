import React from 'react';
import { mount } from "enzyme"
import { Provider } from "react-redux"

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom'
import { Sidebar } from '../../../components/journal/Sidebar';
import { startLogout } from '../../../actions/auth';
import { startNewNote } from '../../../actions/notes';

jest.mock('../../../actions/auth', () => ({
    startLogout: jest.fn()
}))
jest.mock('../../../actions/notes', () => ({
    startNewNote: jest.fn()
}))

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth: {
        uid: '1',
        name: 'Pablo',

    },
    ui: {
        loading: false,
        msgError: null
    },
    notes:{
        active:null,
        notes:[]
    }
};

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <Sidebar/>
    </Provider>
);


describe('Pruebas en el <Sidebar />', () => {

    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de llamar la accion del logout', () => {
        wrapper.find('button').prop('onClick')();

        expect(startLogout).toHaveBeenCalled();
    })

    test('Debe de llamar al startNewNote', () => {
        
        wrapper.find(".journal__new-entry").prop('onClick')();

        expect(startNewNote).toHaveBeenCalled();

    })

})