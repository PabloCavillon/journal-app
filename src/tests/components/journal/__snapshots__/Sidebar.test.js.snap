// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <Sidebar /> Debe de mostrarse correctamente 1`] = `
<aside
  className="journal__sidebar"
>
  <div
    className="jounal__sidebar-navbar"
  >
    <h3
      className="mt-5"
    >
      <i
        className="far fa-moon"
      />
      <span>
         
        Pablo
      </span>
    </h3>
    <button
      className="btn"
      onClick={[Function]}
    >
      logout
    </button>
  </div>
  <div
    className="journal__new-entry"
    onClick={[Function]}
  >
    <i
      className="far fa-calendar-plus fa-5x"
    />
    <p
      className="mt-5"
    >
      New entry
    </p>
  </div>
  <div
    className="journal__entries"
  />
</aside>
`;
